### DISCUSSION TITLE
cookiecutter with poetry

### DISCUSSION THREAD

**Discussion by lecrami** [python-poetry/poetry#6851] [General]
i am trying to use Cookiecutter with django template by pydanny. Im using poetry to manage my python project and i don't know how to do it.

Normally using pip and a virtual environment, all required packages are installed with:

` pip install -r requirements/local.txt`

But in poetry i can't install dependencies from '.txt' file. What do I have to do?

**Comment by danmur** [comment]
Poetry can export a requirements.txt but as far as I know it can't import one. If it were me I'd just call poetry add ... for each of the requirements by hand, or if there were a lot, script it. You only need to do it once.

**Comment by hemanta212** [comment]
Related #663  and #46

It's highly unlikely to be added.

**Comment by etijskens** [comment]
FYI:
https://github.com/etijskens/et-micc
https://micc.readthedocs.io/en/latest/

On 2 Mar 2020, at 10:52, Hemanta <notifications@github.com<mailto:notifications@github.com>> wrote:


Related #663<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpython-poetry%2Fpoetry%2Fissues%2F663&data=02%7C01%7Cengelbert.tijskens%40uantwerpen.be%7C379d2fbdd1dc46377a7308d7be8f6276%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637187395348868633&sdata=DH9qmouulZpa8smn2rGcfoIfhepJb8iUES8FWLo2QjA%3D&reserved=0> and #46<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpython-poetry%2Fpoetry%2Fissues%2F46&data=02%7C01%7Cengelbert.tijskens%40uantwerpen.be%7C379d2fbdd1dc46377a7308d7be8f6276%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637187395348868633&sdata=NYTcRyL%2FvAfao5WEU83yfmiJcnnKgL4y7fEHxALjbao%3D&reserved=0>

It's highly unlikely to be added.

**Comment by danmur** [comment]
Hey I lost this issue, but here it is again! dephell might do the job (https://dephell.org/).

**Comment by SanketDG** [comment]
After going through #663 and #2135, and realizing this is the only issue that is open, I think poetry should atleast provide some direction for new users when they are converting from other formats.

Let's just say `requirements.txt`  is the most ubiquitous form of specifying requirements in the Python ecosystem.

Not providing any information on this would stray away a lot of new users.

It could be as simple as redirecting to https://dephell.readthedocs.io/cmd-deps-convert.html in the documentation. This needs to happen in the least.

Maintainers of poetry, please provide some insight on this.

**Comment by src-r-r** [comment]
Dephell is archived, unfortunately.

I'll admit I got a little impatient, so I created this Python tool to do just this (called stanza...get it?). Parses requirements.txt files and automatically names and versions your project. It even allows for referenced requirements "e.g. `-r ...`"

https://github.com/src-r-r/python-stanza

I created it in a rush (but still tested it), so contributions welcome!
